portfolio/src/main/java/com/google/sps/servlets/LoginServlet.java

```diff
@@ -30,29 +35,58 @@
 
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    // If user is logged in and has not set a nickname, redirect to nickname page
     UserService userService = UserServiceFactory.getUserService();
+    if (userService.isUserLoggedIn()) {
+      
+    }
+    
+    // Create variables needed to create a user.
     Boolean loggedIn = false;
     String authenticationUrl = ""; // URL used for either login or logout.
     String userEmail = "";
+    String nickname = "";
     String redirectUrl = "/"; // Both login and logout redirect to the same URL
 
+    // Define values for user variables depending on login status.
     if (userService.isUserLoggedIn()) {
-      // Create a new user with logout URL and email.
       loggedIn = true;
-      authenticationUrl = userService.createLogoutURL(redirectUrl);
       userEmail = userService.getCurrentUser().getEmail();
+      nickname = getUserNickname(userService.getCurrentUser().getUserId());
+      if (nickname == null) {
+        // If logged in user has no nickname, redirect to nickname setup page.
+        nickname = "";
+        authenticationUrl = "/nickname.html";
+      } else {
+        // If logged in user has nickname, set login URL.
```

Comment:
nit: I think this should say "If logged in user has a nickname, set logout URL."

Reply:
Done!